app/models/collection_participant.rb

```diff
@@ -57,4 +59,9 @@ def user_allowed_to_promote?(user, role)
     (role == MEMBER || role == NONE) ? self.collection.user_is_maintainer?(user) : self.collection.user_is_owner?(user)
   end
 
+  def reindex_collection
+    return unless MAINTAINER_ROLES.include?(participant_role)
```

Comment:
Would you be able to explain this a little bit more? I'm not certain exactly what the change you're asking for looks like.

Reply:
I think ticking means we need to reindex the collection if
* The role changes from member/invited/none to either owner or moderator. In this case, reindexing will add the user's id to the maintainer_ids array and and either the owner_ids or moderator_ids array.
* The role changes from moderator/owner to member, invited, or none. In this case, reindexing will remove the user's id from the maintainer_ids array and either the owner_ids or moderator_ids array.
* The role changes from moderator to owner or vice versa. In this case, reindexing will move the user's id out of moderator_ids and into owner_ids (or vice versa).

Reply:
Yep. The code here covers the first and third cases (because in both of those cases, the participant role after the update is included in `MAINTAINER_ROLES`), but not the second (because in that case, the participant role is included in `MAINTAINER_ROLES` only before the update, not after, so it won't make it past this guard statement).